### PR TITLE
sqlalchemy 1.4 support for sqlalchemy-roles-preview

### PR DESCRIPTION
**Pull request by saolsen**
Using a sqlalchemy declarative Base is still required but it doesn't break on 1.4 anymore.

### CODE REVIEW

**Comment by saolsen** [comment]
This pr also includes a bunch of stuff that was on main and not merged in to sqlalchemy-roles-preview yet because the tests weren't passing. This merge will get it back up to date.

**Comment by saolsen** [comment]
The relevant commit to review is https://github.com/osohq/oso/pull/932/commits/ae614bc8a44ec99c22244523e70ddec738c05710

**Comment by samscott89** [comment]
> 
> 
> The relevant commit to review is [ae614bc](https://github.com/osohq/oso/commit/ae614bc8a44ec99c22244523e70ddec738c05710)

Cool, this looks good. Will try it out locally later and try it out.

Also, there's the failing docs job to solve.

File: languages/python/sqlalchemy-oso/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-oso==0.12.2
-sqlalchemy~=1.3.20
+oso>=0.12.2
```
Comment (gj) [inline languages/python/sqlalchemy-oso/requirements.txt:1]:
```suggestion
oso~=0.12.3
```

**Review by samscott89 — APPROVED** [review]
Checked the non-merge related changes. They look good.

**Pull request merged**